### ISSUE TITLE
Support ubuntu 18.04 Bionic?

### ISSUE THREAD

**Issue by jeff1985** [geerlingguy/ansible-role-docker#60]
Hi,

I have tried to use this repo with ubuntu 18.04, but it fails. 
Also I have seen [this commit](https://github.com/geerlingguy/ansible-role-docker/commit/201cd7dd947541c3e1cc697fca41c42087524972) declaring that its not compatible.
 
Are you planning to add support for Ubuntu 18.04 in the next time?

**Comment by geerlingguy** [comment]
Currently it looks like the Docker repo doesn't have packages available for Bionic; see https://github.com/docker/for-linux/issues/290#issuecomment-387282484

Hopefully this will be remedied soon. For now they recommend using the Docker installer script (which is not easy to automate with idempotence and doesn't (IIRC) use a package like this role does (so upgrading is... weird)).

I'll leave this ticket open until hopefully early June, when the next CE stable release is out, and it should end up in the Bionic repo.

**Comment by frankjoshua** [comment]
You can use the edge version instead of stable until stable is released.
`
    docker_apt_release_channel: edge
`
Maybe better than nothing.

**Comment by brainstorm** [comment]
After activating the edge channel, I [had to unmask the service for it to work properly](http://www.pesikov.tk/blog/failed-restart-docker-service-unit-docker-service-masked/), otherwise I got:

```
fatal: [default]: FAILED! => {"changed": false, "msg": "Error connecting: Error while fetching server API version: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))"}
```

TL;DR:

```bash
systemctl unmask docker.service
systemctl unmask docker.socket
```

**Comment by brainstorm** [comment]
Not even with the unmasking, the docker client is just too old on `edge`:

```bash
root@localhost:/home/vagrant# docker --version
Docker version 17.12.1-ce, build 7390fc6
```

Here's the ansible error:

```python
The full traceback is:
  File "/tmp/ansible_ztsh4zex/ansible_modlib.zip/ansible/module_utils/docker_common.py", line 182, in __init__
    super(AnsibleDockerClient, self).__init__(**self._connect_params)
  File "/usr/local/lib/python3.6/dist-packages/docker/client.py", line 99, in __init__
    self._version = self._retrieve_server_version()
  File "/usr/local/lib/python3.6/dist-packages/docker/client.py", line 124, in _retrieve_server_version
    'Error while fetching server API version: {0}'.format(e)
```

**Comment by geerlingguy** [comment]
Yeah... sadly I'm not planning on 'official' support until the stable channel has a new release. C'mon Docker, let's do it :D

**Comment by geerlingguy** [comment]
Keep watching https://github.com/docker/for-linux/issues/290#issuecomment-387282484, I'm sure one of the many followers on that topic will comment once it's out!

**Comment by geerlingguy** [comment]
Running some CI tests now that the issue linked above looks like it may be resolved!

**Comment by geerlingguy** [comment]
It works!

**Comment by jeff1985** [comment]
Thanks a lot!
